src/main/sensors/battery.c

```diff
@@ -232,16 +233,48 @@ static void batteryUpdateVoltageState(void)
         default:
             break;
     }
+
+}
+
+static void batteryUpdateLVC(timeUs_t currentTimeUs) 
+{   
+    if (batteryConfig()->lvcEnabled) {
+        if (voltageState == BATTERY_CRITICAL && !lowVoltageCutoff.enabled) {
+            lowVoltageCutoff.enabled = true;
+            lowVoltageCutoff.startTime = currentTimeUs;
+            lowVoltageCutoff.percentage = 100;
+        }
+        if (lowVoltageCutoff.enabled) {
+            if (currentTimeUs - lowVoltageCutoff.startTime < LVC_AFFECT_TIME) {
+                lowVoltageCutoff.percentage = 100 - ((currentTimeUs - lowVoltageCutoff.startTime) * (100 - batteryConfig()->lvcPercentage) / LVC_AFFECT_TIME); 
```

Comment:
`cmp32(currentTimeUs, lowVoltageCutoff.startTime)` will handle userc overflow correctly
Also maximum `LVC_AFFECT_TIME` is 43s for unsigned variant ant 21s for signed ... there is quite little safety margin